Plugins/Public/base_plugin/FactoryModule.cpp

```diff
@@ -32,53 +25,95 @@ wstring FactoryModule::GetInfo(bool xml)
 	if (Paused)	Status = L"(Paused) ";
 	else Status = L"(Active) ";
 
+	info += recipeMap[factoryNickname].infotext;
+
+	wstring openLine;
+	wstring closeLine;
 	if (xml)
 	{
-		info += FACTORY_NAMES[type];
-		info += L"</TEXT><PARA/><TEXT>      Pending " + stows(itos(build_queue.size())) + L" items</TEXT>";
-		if (active_recipe.nickname)
+		openLine = L"</TEXT><PARA/><TEXT>      ";
+		closeLine = L"</TEXT>";
+	}
+	else
+	{
+		openLine = L" - ";
+		closeLine = L"\n";
+	}
+	info += openLine + L"Pending " + stows(itos(build_queue.size())) + L" items";
+	if (active_recipe.nickname)
+	{
+		info += openLine + L"Crafting " + Status + active_recipe.infotext + L". Waiting for:";
+
+		for (auto& i : active_recipe.consumed_items)
 		{
-			info += L"<PARA/><TEXT>      Building " + Status + active_recipe.infotext + L". Waiting for:</TEXT>";
+			uint good = i.first;
+			uint quantity = i.second;
 
-			for (map<uint, uint>::iterator i = active_recipe.consumed_items.begin();
-				i != active_recipe.consumed_items.end(); ++i)
+			const GoodInfo* gi = GoodList::find_by_id(good);
+			if (gi)
+			{
+				info += openLine + L"- " + stows(itos(quantity)) + L"x " + HkGetWStringFromIDS(gi->iIDSName);
+				if (quantity > 0 && base->HasMarketItem(good) < active_recipe.cooking_rate)
+				{
+					info += L" [Out of stock]";
+				}
+			}
+		}
+		if (active_recipe.credit_cost)
+		{
+			info += openLine + L" - Credits x" + stows(itos(active_recipe.credit_cost));
```

Comment:
ditto comments about both closeLine and credits formatting

Reply:
See above and added credit formatting